twin.go

```diff
@@ -115,13 +165,23 @@ func (s *Substrate) CreateTwin(identity Identity, ip net.IP) (uint32, error) {
 }
 
 // UpdateTwin updates a twin
-func (s *Substrate) UpdateTwin(identity Identity, ip net.IP) (uint32, error) {
+func (s *Substrate) UpdateTwin(identity Identity, relay *string, pk *string) (uint32, error) {
 	cl, meta, err := s.getClient()
 	if err != nil {
 		return 0, err
 	}
 
-	c, err := types.NewCall(meta, "TfgridModule.update_twin", ip.String())
+	relay_bytes := types.OptionBytes{}
+	if relay != nil {
```

Comment:
improve the validation on the relay and pk in the very top of the function please and terminate early with proper errors.